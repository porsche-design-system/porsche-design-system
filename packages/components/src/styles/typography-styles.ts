import type { JssStyle } from 'jss';
import type { TextAlign, TextColor, Theme } from '../types';
import type { HeadingAlign, HeadingColor } from '../components/heading/heading-utils';
import type { HeadlineColor } from '../components/headline/headline-utils';
import type { DisplayAlign, DisplayColor } from '../components/display/display-utils';
import type { TextColorDeprecated } from '../components/text/text-color';
import { getThemedTypographyColor } from './text-icon-styles';

export const getTypographyRootJssStyle = (
  baseTextStyle: JssStyle,
  responsiveStyle: JssStyle,
  align: TextAlign | HeadingAlign | DisplayAlign,
  color: Exclude<TextColor, TextColorDeprecated> | HeadlineColor | HeadingColor | DisplayColor,
  ellipsis: boolean,
  theme: Theme
): JssStyle => {
  return {
    margin: 0,
    padding: 0,
    ...baseTextStyle,
    color: getThemedTypographyColor(theme, color),
    textAlign: align,
    letterSpacing: 'normal',
    listStyleType: 'none',
    whiteSpace: 'inherit',
    ...(ellipsis && {
      maxWidth: '100%',
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      whiteSpace: 'nowrap',
    }),
    ...responsiveStyle,
  };
};

export const getTypographySlottedJssStyle = (): JssStyle => {
  return {
    margin: 'inherit',
    padding: 'inherit',
    fontFamily: 'inherit',
    fontWeight: 'inherit',
    fontSize: 'inherit',
    lineHeight: 'inherit',
    fontStyle: 'inherit',
    fontVariant: 'inherit',
    color: 'inherit',
    textAlign: 'inherit',
    overflowWrap: 'inherit',
    wordWrap: 'inherit',
    hyphens: 'inherit',
    whiteSpace: 'inherit',
    letterSpacing: 'inherit',
  };
};
