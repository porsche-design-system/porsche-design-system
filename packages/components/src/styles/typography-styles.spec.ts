import { getEllipsisStyle, getSlottedTypographyStyle } from './typography-styles';

describe('getSlottedTypographyStyle()', () => {
  it('should return correct JssStyle', () => {
    expect(getSlottedTypographyStyle()).toMatchSnapshot();
  });
});

describe('getEllipsisStyle()', () => {
  it('should return correct JssStyle', () => {
    expect(getEllipsisStyle()).toMatchSnapshot();
  });
});
