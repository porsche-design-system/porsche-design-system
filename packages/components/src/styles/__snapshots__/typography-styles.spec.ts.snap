// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getEllipsisJssStyle() should return correct JssStyle 1`] = `
{
  "maxWidth": "100%",
  "overflow": "hidden",
  "textOverflow": "ellipsis",
  "whiteSpace": "nowrap",
}
`;

exports[`getSlottedTypographyStyle() should return correct JssStyle 1`] = `
{
  "color": "inherit",
  "fontFamily": "inherit",
  "fontSize": "inherit",
  "fontStyle": "inherit",
  "fontVariant": "inherit",
  "fontWeight": "inherit",
  "hyphens": "inherit",
  "lineHeight": "inherit",
  "margin": "inherit",
  "overflowWrap": "inherit",
  "padding": "inherit",
  "textAlign": "inherit",
  "whiteSpace": "inherit",
  "wordWrap": "inherit",
}
`;
